code/analytics/src/test/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsExtensionTests.kt

```diff
@@ -0,0 +1,162 @@
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.analytics.internal
+
+import com.adobe.marketing.mobile.EventSource
+import com.adobe.marketing.mobile.EventType
+import com.adobe.marketing.mobile.ExtensionApi
+import com.adobe.marketing.mobile.ExtensionEventListener
+import com.adobe.marketing.mobile.ExtensionHelper
+import com.adobe.marketing.mobile.services.DataStoring
+import com.adobe.marketing.mobile.services.NamedCollection
+import com.adobe.marketing.mobile.services.ServiceProvider
+import com.adobe.marketing.mobile.util.SQLiteUtils
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.ArgumentCaptor
+import org.mockito.Mock
+import org.mockito.Mockito.any
+import org.mockito.Mockito.eq
+import org.mockito.Mockito.mockStatic
+import org.mockito.Mockito.reset
+import org.mockito.Mockito.times
+import org.mockito.Mockito.verify
+import org.mockito.Mockito.`when`
+import org.mockito.junit.MockitoJUnitRunner
+
+@RunWith(MockitoJUnitRunner.Silent::class)
+class AnalyticsExtensionTests {
+    private lateinit var analytics: AnalyticsExtension
+
+    @Mock
+    private lateinit var mockApi: ExtensionApi
+
+    @Mock
+    private lateinit var mockDataStore: DataStoring
+
+    @Mock
+    private lateinit var mockNamedCollection: NamedCollection
+
+    @Mock
+    private lateinit var mockDatabase: AnalyticsDatabase
+
+    @Mock
+    private lateinit var mockServiceProvider: ServiceProvider
+
+    private val mockedStaticServiceProvider = mockStatic(
+        ServiceProvider::class.java
+    )
+
+    @Before
+    fun setup() {
+        mockedStaticServiceProvider.`when`<Any> { ServiceProvider.getInstance() }
+            .thenReturn(mockServiceProvider)
+        `when`(mockDataStore.getNamedCollection(eq("AnalyticsDataStorage"))).thenReturn(mockNamedCollection)
+        `when`(mockServiceProvider.dataStoreService).thenReturn(mockDataStore)
+        analytics = AnalyticsExtension(mockApi, mockDatabase)
+    }
+
+    @After
+    fun tearDown() {
+        reset(mockApi)
+        reset(mockDataStore)
+        reset(mockDatabase)
+        reset(mockNamedCollection)
+        mockedStaticServiceProvider.close()
+        reset(mockServiceProvider)
+    }
+
+    @Test
+    fun `onRegistered() should call createSharedState for version 0`() {
```

Comment:
Can you add another test to verify an empty shared state is created if there is no AID for VID values?  I know the code always creates the shared state now, but if the code is updated in the future it is good to have a test which asserts an empty state is created.